src/app/core/components/change-password/change-password.component.html

```diff
@@ -0,0 +1,94 @@
+<div class="headline-button">
+  <h2>Password</h2>
+  <div class="options-headline">
+    <p-button
+      *ngIf="passwordFormControl.disabled; else passwordEnabled"
+      label="Change"
+      styleClass="p-button-text p-button-sm"
+      icon="pi pi-pencil"
+      (click)="togglePasswordDisabled()" />
+    <ng-template #passwordEnabled>
+      <div class="enabled-options">
+        <p-button
+          label="Discard"
+          styleClass="p-button-text p-button-sm"
+          icon="pi pi-times"
+          (click)="togglePasswordDisabled()" />
+        <p-button
+          label="Update"
+          styleClass="p-button-sm"
+          [disabled]="!passwordFormControl.valid"
+          icon="pi pi-check"
+          (click)="updatePassword()" />
+      </div>
+    </ng-template>
+  </div>
+</div>
+<form [formGroup]="passwordFormControl" class="input-group">
+  <div class="input-wrapper">
+    <input
+      pInputText
+      placeholder="Original password"
+      type="password"
+      id="originalPassword"
+      formControlName="originalPassword" />
+    <small
+      class="validator-error"
+      *ngIf="
+        passwordFormControl.controls['originalPassword'].errors?.[
+          'wrongPassword'
+        ]
+      ">
+      Incorrect Password
```

Comment:
that was spooky. did not expect that someone tells me that this password is not correct.
I have never seen this and I'm thinking that might be caused by some security implications. Fine for your app but I don't know if such a checking endpoint would be good for applications in production. Just be careful with things like passwords and personal innovations 🙏